compiler/semmagic.nim

```diff
@@ -190,9 +190,20 @@ proc evalTypeTrait(c: PContext; traitCall: PNode, operand: PType, context: PSym)
     result = newIntNodeT(toInt128(operand.len), traitCall, c.idgen, c.graph)
   of "distinctBase":
     var arg = operand.skipTypes({tyGenericInst})
-    while arg.kind == tyDistinct:
+    let rec =
+      if traitCall.len >= 2:
+        operand2.intVal != 0
+      else:
+        true
```

Comment:
Is this really the best way to do this? Can't this read the `static bool` directly somehow?

Reply:
I'm pretty sure that it is (also see my last change, replaced `operand2` with `traitCall[2]`)

Reply:
this code is incorrect, and fails